### ISSUE TITLE
k8sattributes processor not allowing metadata

### ISSUE THREAD

**Issue by v1ctorrhs** [open-telemetry/opentelemetry-collector-contrib#21228]
### Component(s)

processor/k8sattributes

## Description

According to k8sattributes processor [README.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/README.md) L64 and [config.goL78 
](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/config.go#L78) point that if container.id is provided k8s.container.name will be available as an attribute.

However, I can't get that to work as expected.
## Steps to Reproduce
k8sattributes config
```
      k8sattributes:
        passthrough: false
        auth_type: "serviceAccount"
        extract:
          metadata:
          - k8s.pod.name
          - k8s.pod.uid
          - k8s.deployment.name
          - k8s.namespace.name
          - k8s.node.name
          - k8s.pod.start_time
          - container.id
          labels:
          # extracts value of label from pods with key `app.kubernetes.io/version` and inserts it as a tag with key `service.version`
          - tag_name: service.version
            key: app.kubernetes.io/version
            from: pod
        pod_association:
          - sources:
            - from: connection
          - sources:
            - from: resource_attribute
              name: k8s.pod.ip
          - sources:
            - from: resource_attribute
              name: k8s.pod.uid
          # below association matches for pair `k8s.pod.name` and `k8s.namespace.name`
          - sources:
            - from: resource_attribute
              name: k8s.pod.name
            - from: resource_attribute
              name: k8s.namespace.name
```
This config returns the following attributes
<img width="578" alt="image" src="https://user-images.githubusercontent.com/67688658/234913747-e28cb487-2bdb-4a4f-9933-464da0918632.png">

If i try to add
```
extract:
 metadata:
 ...
 ...
 - k8s.container.name
``` 
I get the following error

>2023-04-27T15:30:47.009Z	debug	processor/processor.go:298	Beta component. May change in the future.	{"kind": "processor", "name": "resource", "pipeline": "traces"}
2023-04-27T15:30:47.009Z	debug	processor/processor.go:298	Beta component. May change in the future.	{"kind": "processor", "name": "k8sattributes", "pipeline": "traces"}
Error: failed to build pipelines: failed to create "k8sattributes" processor, in pipeline "traces": "k8s.container.name" is not a supported metadata field
2023/04/27 15:30:47 collector server run finished with error: failed to build pipelines: failed to create "k8sattributes" processor, in pipeline "traces": "k8s.container.name" is not a supported metadata field

### Collector version

v0.75.0

**Comment by v1ctorrhs** [comment]
hmm, looks like this was added after v0.75.0 -> here https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/21591e14ef777f69711403000c9169a56be3ecb9

**Comment by dmitryax** [comment]
@gbbr can you please take a look?

**Comment by gbbr** [comment]
Hi @v1ctorrhs 👋🏻 

> how did you expose the container.ID info inside the k8s container env? Looking at the k8s [docs](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables) we can play with status.containerStatuses and all the ID's are available there, however that returns a list of all IDs.

The correct approach is to send the container ID from the container itself, from within the app. For example, in Go, the `WithContainer` resource option [from the SDK](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.14.0/resource#WithContainer) allows you to do this automatically and works well. However, I notice that this SDK option only works with cgroups v1.

> I get the following error [...]

I tried your setup, including the addition of `k8s.container.name` to the metadata and did not get any errors. Do you mind trying again perhaps with >0.76.1? 

In the meantime, I am also trying to investigate deeper.

**Comment by gbbr** [comment]
I suspect we may be able to close this issue, since you were expecting to see functionality that's only available in v0.76.3, in v0.75.0. Perhaps you can confirm @v1ctorrhs
